activerecord/test/cases/bind_parameter_test.rb

```diff
@@ -238,10 +238,11 @@ def assert_bind_params_to_sql
           #
           #   SELECT `authors`.* FROM `authors` WHERE `authors`.`id` IN (1, 2, 3)
           #
-          if current_adapter?(:Mysql2Adapter)
-            params = bind_params((1..3).map(&:to_s))
+          params = if current_adapter?(:Mysql2Adapter) || current_adapter?(:TrilogyAdapter)
```

Comment:
I believe `:AbstractMysqlAdapter` would actually work here... but that's not our current established pattern. When I eventually find a way to separate the "communication library interface" and "server SQL dialect" concepts, a more general notion of "this applies to adapters that speak MySQL-ish" would definitely make many of these conditionals read more pleasantly [and more contextually].

Reply:
Trilogy should never run this test file since it doesn't enable (or support) prepared statements yet:
https://github.com/rails/rails/pull/53702#discussion_r1858000512